File: frontend/src/types/document.ts
export interface Document {
  id: string;
  title: string;
  template_slug: string;
  category: string;
  department: string;
  input_data: Record<string, unknown>;
  ai_data: Record<string, unknown>;
  file_path: string;
  onec_status: string;
  onec_doc_id?: string;
  created_at: string;
}

export interface DocumentListItem {
  id: string;
  title: string;
  template_slug: string;
  category: string;
  department: string;
  onec_status: string;
  created_at: string;
}

File: backend/app/models/document.py
import uuid
from datetime import datetime

from sqlalchemy import String, DateTime, Text, ForeignKey, func
from sqlalchemy.dialects.postgresql import UUID, JSONB
from sqlalchemy.orm import Mapped, mapped_column

from app.database import Base


class Document(Base):
    __tablename__ = "documents"

    id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    template_slug: Mapped[str] = mapped_column(String(100), nullable=False)
    title: Mapped[str] = mapped_column(String(255), nullable=False)
    category: Mapped[str] = mapped_column(String(100), default="")
    department: Mapped[str] = mapped_column(String(50), nullable=False)
    user_id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), ForeignKey("users.id"), nullable=False)
    input_data: Mapped[dict] = mapped_column(JSONB, default=dict)
    ai_data: Mapped[dict] = mapped_column(JSONB, default=dict)
    file_path: Mapped[str] = mapped_column(String(500), default="")
    onec_doc_id: Mapped[str] = mapped_column(String(100), default="")
    onec_status: Mapped[str] = mapped_column(String(50), default="pending")
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now())
